qiskit_ibm_provider/ibm_backend.py

```diff
@@ -457,90 +456,117 @@ def run(
+        if init_qubits is not None:
+            inputs["init_qubits"] = init_qubits
+        if parameter_binds:
+            inputs["parameter_binds"] = parameter_binds
+        if use_measure_esp is not None:
+            inputs["use_measure_esp"] = use_measure_esp
+        if live_data_enabled is not None:
+            inputs["live_data_enabled"] = live_data_enabled
+        if noise_model:
+            inputs["noise_model"] = noise_model
+        if run_config:
+            for key, value in run_config.items():
+                inputs[key] = value
 
-        run_config_dict = self._get_run_config(
-            qobj_header=header,
-            shots=shots,
-            memory=memory,
-            qubit_lo_freq=qubit_lo_freq,
-            meas_lo_freq=meas_lo_freq,
-            schedule_los=schedule_los,
-            meas_level=meas_level,
-            meas_return=meas_return,
-            memory_slots=memory_slots,
-            memory_slot_size=memory_slot_size,
-            rep_time=rep_time,
-            rep_delay=rep_delay,
-            init_qubits=init_qubits,
-            use_measure_esp=use_measure_esp,
-            noise_model=noise_model,
-            **run_config,
+        return self._runtime_run(
+            program_id=program_id, inputs=inputs, options=options, job_tags=job_tags
         )
-        if parameter_binds:
-            run_config_dict["parameter_binds"] = parameter_binds
-        if sim_method and "method" not in run_config_dict:
-            run_config_dict["method"] = sim_method
-
-        if isinstance(circuits, list):
-            chunk_size = None
-            if hasattr(self.configuration(), "max_experiments"):
-                backend_max = self.configuration().max_experiments
-                chunk_size = (
-                    backend_max
-                    if max_circuits_per_job is None
-                    else min(backend_max, max_circuits_per_job)
-                )
-            elif max_circuits_per_job:
-                chunk_size = max_circuits_per_job
-
-            if chunk_size and len(circuits) > chunk_size:
-                circuits_list = [
-                    circuits[x : x + chunk_size]
-                    for x in range(0, len(circuits), chunk_size)
-                ]
-                return IBMCompositeJob(
-                    backend=self,
-                    api_client=self._api_client,
-                    circuits_list=circuits_list,
-                    run_config=run_config_dict,
-                    name=job_name,
-                    tags=job_tags,
-                )
 
-        qobj = assemble(circuits, self, **run_config_dict)
+    def _runtime_run(
+        self,
+        program_id: str,
+        inputs: Dict,
+        options: Dict,
+        job_tags: Optional[List[str]] = None,
+    ) -> IBMCircuitJob:
+        """Runs the runtime program and returns the corresponding job object"""
+        hgp = self.provider._get_hgp(backend_name=options["backend"])
```

Comment:
How is this currently being done? I didn't find it in the old `backend.run()` method. It seems that currently the provider does not explicitly save the instance it was given.

Reply:
It wouldn't return a random hgp right? `_get_hgp` always returns the first hgp when given a backend and that should be the initialized hgp

Reply:
I think you are right if I initialized the provider with a hgp. But it doesn't look like `IBMBackend` has a hgp attribute, so this would still return a random backend if I do 

```
provider = IBMProvider()
backend = provider.get_backend(instance="h1/g1/p1")
backend.run()  <--- I think this would use a random backend?
```

Reply:
it wouldn't choose a random backend - if only the instance is given then it would raise an error if there are multiple backends 

https://github.com/Qiskit/qiskit-ibm-provider/blob/main/qiskit_ibm_provider/ibm_provider.py#L724

Reply:
Sorry I meant a random h/g/p, not a random backend. If I specify a h/g/p in `get_backend()`, then the subsequent `backend.run()` should use that h/g/p.

Reply:
I believe the `get_backend()` method has always just returned a backend that matches the filters

But yes, I agree, if the instance is passed in `get_backend()`, that instance should be used - will look into how to  fix this